doc/ongoing_operations.rst

```diff
@@ -81,34 +81,72 @@ Perform action on a set of jobs
 -------------------------------
 
 Sometimes you want to do bulk actions like disable or delete all jobs of a specific distro or just one target.
-We recommend running a Groovy scripts using the script console.
+We recommend running a Groovy script using the script console.
+
+You can change the distro letter or add or remove prefixes from the list in order to reduce the scope of targeted jobs.
+
+Some operations take time and an HTTP timeout creates performance issues with dangling script runs.
+To prevent them the example script uses a ``BATCH_SIZE`` constant to control how many jobs to change before returning.
+You can leave the default, fairly conservative batch size or increase that constant steadily until results are no longer instantaneous and then adjusting back down.
+Running the script repeatedly until there are no longer any remaining jobs to process will work as long as non-destructive changes, like enabling or disabling jobs, are properly skipped.
 
 The following Groovy script is a good starting point for various actions:
 
 .. code-block:: groovy
 
-   import hudson.model.Cause
+   DISTRO_LETTER = "F"
+   PREFIXES = ["ci_", "dev_", "pr_", "rel", "src_", "bin_"].collect({pre -> DISTRO_LETTER + pre})
```

Comment:
It might be overkill to add a DISTRO_LETTER constant one line before its only use but I like making the bits that are meant to be adjusted very obvious rather than just hard-coding the letter into the map/collect that generates all of the prefixes.

Reply:
Just as an FYI, if we wanted to get just a bit fancy, we could do this at the top:

```
DISTRO_NAME = "foxy"
DISTRO_LETTER = DISTRO_NAME[0..0].toUpperCase()
PREFIXES = [DISTRO_NAME]
PREFIXES += ["ci_", "dev_", "pr_", "rel", "src_", "bin_"].collect({pre -> DISTRO_LETTER + pre})   
```

That way we'll pick up the jobs that are prefixed with the `DISTRO_NAME` as well.

Reply:
FYI, this is also missing the `doc` jobs, so it should also be added to the list above.